src/app/ApplicationTemplate.Shared/Client/Data/Http/HttpLog.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+
+namespace ApplicationTemplate.Client
+{
+	public record HttpLog
+	{
+		public HttpLog(HttpRequestContent request, HttpResponseContent response = null, Status status = Status.Sent)
+		{
+			Request = request;
+			Response = response;
+			CurrentStatus = status;
+		}
+
+		public Guid Id { get; } = Guid.NewGuid();
+
+		public HttpRequestContent Request { get; }
+
+		public HttpResponseContent Response { get; }
+
+		public Status CurrentStatus { get; }
+
+		/// <summary>
+		/// Used to know the the processing status of the HttpRequest.
+		/// </summary>
+		public enum Status
+		{
+			/// <summary>
+			/// Describes the request as being sent.
+			/// </summary>
+			Sent,
+			/// <summary>
+			/// Describes the response as being received.
+			/// </summary>
+			Received,
```

Comment:
If you receive an http 400, is it Received or Failed?

Reply:
We get a received status, this is just the state of the response, whether or not we received the response or failed to receive the response. The StatusCode will still be available to know which state the response is in once it's received.